api/http/handler/hostmanagement/intel/amtrpc.go

```diff
@@ -0,0 +1,222 @@
+
+	// TODO: start
+	//      docker run --rm -it --privileged ptrrd/openamt:rpc-go amtinfo
+	//		on the Docker standalone node (one per env :)
+	//		and later, on the specified node in the swarm, or kube.
+	nodeName := ""
+	docker, err := handler.DockerClientFactory.CreateClient(endpoint, nodeName)
+	if err != nil {
+		return &httperror.HandlerError{StatusCode: http.StatusInternalServerError, Message: "Unable to find an endpoint with the specified identifier inside the database", Err: err}
+	}
+	defer docker.Close()
+
+	ctx := context.TODO()
+	// pull the image so we can check if there's a new one
+	imagename := "ptrrd/openamt:rpc-go"
+	containerName := "openamt-rpc-go"
+
+	if err := pullImage(ctx, docker, imagename); err != nil {
+		return &httperror.HandlerError{StatusCode: http.StatusInternalServerError,
+			Message: "Could not pull image from registry", Err: err}
+	}
+
+	output, err := runContainer(ctx, docker, imagename, containerName, []string{"amtinfo"})
+	if err != nil {
+		return &httperror.HandlerError{StatusCode: http.StatusInternalServerError,
+			Message: "Could not run container", Err: err}
+	}
+
+	amtInfo := Info{
+		Endpoint: portainer.EndpointID(endpointID),
+		Text:     output,
+	}
+	return response.JSON(w, amtInfo)
+}
+
+// TODO: ideally, pullImage and runContainer will become a simple version of the use compose abstraction that can be called from withing Portainer.
+// TODO: the idea being that if we have an internal struct of a parsed compose file, we can also populate that struct programatically, and run it to get the result I'm getting here.
+// TODO: likley an upgrade and absrtaction of DeployComposeStack/DeploySwarmStack/DeployKubernetesStack
+// pullImage will pull the image to the specified environment
+// TODO: add k8s implemenation
+// TODO: work out registry auth
+func pullImage(ctx context.Context, docker *client.Client, imagename string) error {
+	r, err := docker.ImagePull(ctx, imagename, types.ImagePullOptions{})
+	if err != nil {
+		logrus.WithError(err).Error("Could not pull %s from registry", imagename)
+		return err
+	}
+	// yeah, swiped this, need to figure out a good way to wait til its done...
+	b := make([]byte, 8)
+	for {
+		_, err := r.Read(b)
+		// TODO: should convert json text to a struct and show just the text messages
+		//if n > 0 {
+		//fmt.Printf(string(b))
+		//}
+		if err == io.EOF {
+			break
+		}
+	}
+	r.Close()
+
+	return nil
+}
+
+// TODO: ideally, pullImage and runContainer will become a simple version of the use compose abstraction that can be called from withing Portainer.
+// runContainer should be used to run a short command that returns information to stdout
+// TODO: add k8s support
+func runContainer(ctx context.Context, docker *client.Client, imagename, containerName string, cmd []string) (output string, err error) {
+	envs := []string{}
+	// for _, envKey := range envKeys {
+	// 	envs = append(envs, options.GetConfigAsEnv(envKey))
+	// }
+	create, err := docker.ContainerCreate(
```

Comment:
doesn't this create a new container each time we call this endpoint?

Reply:
yup, containers are cheap - its really the same thing as a `system.exec` :)